src/SAML2/XML/md/RequestedAttribute.php

```diff
@@ -8,36 +8,40 @@
 use SAML2\Constants;
 use SAML2\Utils;
 use SAML2\XML\saml\Attribute;
+use Webmozart\Assert\Assert;
 
 /**
  * Class representing SAML 2 metadata RequestedAttribute.
  *
  * @package SimpleSAMLphp
  */
-class RequestedAttribute extends Attribute
+class RequestedAttribute extends AbstractMdElement
 {
     /**
      * Whether this attribute is required.
      *
      * @var bool|null
      */
-    private $isRequired = null;
+    protected $isRequired = null;
 
 
     /**
-     * Initialize an RequestedAttribute.
+     * The actual attribute
      *
-     * @param \DOMElement|null $xml The XML element we should load.
+     * @var \SAML2\XML\saml\Attribute
      */
-    public function __construct(DOMElement $xml = null)
-    {
-        parent::__construct($xml);
+    protected $attribute;
 
-        if ($xml === null) {
-            return;
-        }
 
-        $this->isRequired = Utils::parseBoolean($xml, 'isRequired', null);
+    /**
+     * Initialize an RequestedAttribute.
+     *
+     * @param \SAML2\XML\saml\Attribute $attribute
+     */
+    public function __construct(Attribute $attribute, bool $isRequired = null)
```

Comment:
I'm still in doubt whether I should pass an Attribute like I do now, or if I should pass the individual properties and create an Attribute-object in the constructor... I think the latter is the better option, because it would resemble the API of the Attribute-class

Reply:
I just had a look at the standard, and it indeed extends from `saml:Attribute`. Doing the same would make the implementation easier, because we would just need to add an extra boolean property (`isRequired`, defaulting to `null`), getter & setter, and set the namespace and prefix accordingly.